### DISCUSSION TITLE
What's the difference between createMask and createPipe?

### DISCUSSION THREAD

**Discussion by raythurnvoid** [uNmAnNeR/imaskjs#838] [Q&A] [answered]
I don't understand what is the use case that createPipe solves that createMask with resolve can't

**Comment by uNmAnNeR** [comment, marked as answer]
Pipe is stateless. Mask is statefull. Mask usually is used with input elements. Pipe is used to transform one time values.

**Reply by uNmAnNeR** [reply]
Pipe keeps the state of existing mask so same mask can be reused for different scenarios. “Resolve” modifies the state.

**Reply by raythurnvoid** [reply]
Cool thank you!
